modules/programs/xplr.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+with builtins;
+let
+  inherit (lib)
+    types mkIf mkOption mkEnableOption mkPackageOptionMD mdDoc
+    literalExpression;
+
+  cfg = config.programs.xplr;
+  initialConfig = ''
+    version = '${cfg.package.version}'
+  '';
+  # we provide a default version line within the configuration file, which is obtained from the package's attributes
+  # merge the initial configFile, a mapped list of plugins and then the user defined configuration to obtain the final configuration
+  pluginPath = if cfg.plugins != [ ] then
+    (''
+      package.path=
+    '' + (concatStringsSep " ..\n" (map (p: ''"${p}/init.lua;"'') cfg.plugins))
+      + ''
+         ..
+        package.path
+      '')
+  else
+    "\n";
+  configFile = initialConfig + pluginPath + cfg.extraConfig;
+in {
+  meta.maintainers = [ maintainers.notashelf ];
+  options.programs.xplr = {
+    enable = mkEnableOption "xplr, terminal UI based file explorer";
+
+    package = mkPackageOptionMD pkgs "xplr" { };
+
+    plugins = mkOption {
+      type = with types; nullOr (listOf (either package str));
+      default = [ ];
+      defaultText = literalExpression "[]";
+      description = mdDoc ''
+        Plugins to be added to your configuration file. Must be a package, an absolute plugin path, or string
+        to be recognized by xplr. Paths will be relative to $XDG_CONFIG_HOME/xplr/init.lua unless they are absolute.
```

Comment:
keep lines to 80 chars (https://nix-community.github.io/home-manager/index.html#sec-code-style)

Reply:
the most I can truncate it to is roughly 130 characters

"Plugins to be added to your configuration file. Paths will be relative to $XDG_CONFIG_HOME/xplr/init.lua unless they are absolute" 

Any less and some meaning will be lost.

Reply:
Not asking you to shorten the desc, just add line breaks.